src/validation.cpp

```diff
@@ -301,8 +301,8 @@ void Chainstate::MaybeUpdateMempoolForReorg(
     // Iterate disconnectpool in reverse, so that we add transactions
     // back to the mempool starting with the earliest transaction that had
     // been previously seen in a block.
-    auto it = disconnectpool.queuedTx.get<insertion_order>().rbegin();
-    while (it != disconnectpool.queuedTx.get<insertion_order>().rend()) {
+    auto it = disconnectpool.queuedTx.rbegin();
```

Comment:
[this comment](https://github.com/bitcoin/bitcoin/pull/28385/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98L298) still needs updating